domino_data/auth.py

```diff
@@ -28,6 +49,14 @@ def get_headers(self) -> Dict[str, str]:
         if self.api_key:
             return {"X-Domino-Api-Key": self.api_key, **self.headers}
 
+        if self.token_url is not None:
+            try:
+                jwt = get_jwt_token(self.token_url)
+            except httpx.HTTPStatusError:
+                pass
+            else:
+                return {"Authorization": f"Bearer {jwt}", **self.headers}
+
         if self.token_file and exists(self.token_file):
```

Comment:
should we skip this if the former is successful?

Reply:
all my try/except have early return if they are successful. The pass is to avoid using `try..else` without an except which is a bad practice in python (never catch all exception)